### ISSUE TITLE
Error accessing data.json raw github url

### ISSUE THREAD

**Issue by rpj09** [sherlock-project/sherlock#1895]
**Issue Type**: Bug Report

**Summary**: Error accessing data file URL with improved handling

---

### Checklist

Please make sure to check the following boxes if they apply:

- [x] I'm reporting a bug in Sherlock's functionality.
- [x] The bug I'm reporting is not a false positive or a false negative.
- [x] I've verified that I'm running the latest version of Sherlock.
- [x] I've checked for similar bug reports, including closed ones.
- [x] I've checked for pull requests that attempt to fix this bug.

---

### Description

**Bug/Error**: 
```
ERROR:  Problem while attempting to access data file URL 'https://raw.githubusercontent.com/sherlock-project/sherlock/master/sherlock/resources/data.json':  HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Max retries exceeded with url: /sherlock-project/sherlock/master/sherlock/resources/data.json (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x11b7cff70>: Failed to establish a new connection: [Errno 60] Operation timed out'))
```

**Issue Description**:
When attempting to run Sherlock, users encounter the above error message, preventing them from successfully executing Sherlock. This error occurs when Sherlock is unable to access the data file URL ('https://raw.githubusercontent.com/sherlock-project/sherlock/master/sherlock/resources/data.json').

**Comment by rpj09** [comment]
@bharat-nair Yeah, I know that, but in the future, if the issue arises, Sherlock will attempt to load the data.json from the local system, preventing an error. I've also added some additional loggers and visuals to make debugging easier. This not only aids developers but also provides users with a clearer understanding of what Sherlock is doing. Previously, it simply waited for a minute and then gave an error.

**Comment by akh7177** [comment]
Hello @rpj09 ,

Any specific reason for reopening this issue? With the current version of Sherlock there is no longer any issue with accessing data.json from raw github URL and if the user needs to test with the local version of data.json, that can also be done using the _--local_ flag

Thanks!
